extensions/notebook/src/dialog/managePackages/addNewPackageTab.ts

```diff
@@ -234,6 +227,66 @@ export class AddNewPackageTab {
 		});
 	}
 
+	private async fetchCondaPackage(packageName: string): Promise<PipPackageOverview> {
+		let condaExe = this.jupyterInstallation.getCondaExePath();
+		let cmd = `"${condaExe}" search --json ${packageName}`;
+		let packageResult: string;
+		try {
+			packageResult = await this.jupyterInstallation.executeBufferedCommand(cmd);
+		} catch (err) {
+			throw new Error(this.PackageNotFoundError);
+		}
+
+		if (packageResult) {
+			let packageJson = JSON.parse(packageResult);
+			if (packageJson) {
+				if (packageJson.error) {
+					throw new Error(packageJson.error);
+				}
+
+				let packages = packageJson[packageName];
+				if (Array.isArray(packages)) {
+					let allVersions = packages.filter(pkg => pkg && pkg.version).map(pkg => pkg.version);
+					let singletonVersions = new Set<string>(allVersions);
+					let sortedVersions = this.sortPackageVersions(Array.from(singletonVersions));
+					return {
+						name: packageName,
+						versions: sortedVersions,
+						summary: undefined
+					};
+				}
+			}
+		}
+
+		return undefined;
+	}
+
+	private sortPackageVersions(versions: string[]): string[] {
```

Comment:
Can you write some unit tests for this? Just make it public and make sure it's handling things ok?

Reply:
Some of the logic scares me a bit so i'd be interested in how versions like:

0.23.0
2.3
2.30
2.3.0
23
0.23
0.0.23
0.0.0.23
0.0.0.1.23

Reply:
Also I realize that if we get the versioning "wrong" in weird cases it's not the absolute end of the world, but it'd be nice just to have some unit tests regardless